AK/Function.h

```diff
@@ -48,6 +54,19 @@ namespace AK {
 #    define IGNORE_USE_IN_ESCAPING_LAMBDA
 #endif
 
+namespace Detail {
+#if __has_feature(objc_arc)
+inline constexpr bool HaveObjcArc = true;
+#    define BRIDGE_CAST(T, x) ((__bridge T)(x))
+#else
+inline constexpr bool HaveObjcArc = false;
+#    define BRIDGE_CAST(T, x) ((T)(x))
```

Comment:
Errr...why not
```suggestion
template<typename T, typename U> // maybe also some requirements to make this less crazy insane?
static constexpr inline auto bridge_cast(U&& x) -> decltype(auto)
{
    if constexpr (HaveObjcArc)
        return (__bridge T)x;
    else
        return (T)x;
}
```

Reply:
Because both branches of an if constexpr have to be syntactically valid, and `__bridge` isn't a valid type qualifier without ARC enabled.

Reply:
ah. ew (but also let's make this thing a function, even if we use ifdefs inside it)